tests/instruments/lecroy/test_lecroyT3DSO1204.py

```diff
@@ -0,0 +1,367 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import pytest
+
+from pymeasure.instruments.lecroy.lecroyT3DSO1204 import LeCroyT3DSO1204
+from pymeasure.test import expected_protocol
+
+
+def test_init():
+    with expected_protocol(
+            LeCroyT3DSO1204,
+            [("CHDR OFF", None)]
+    ):
+        pass  # Verify the expected communication.
+
+
+def test_bwlimit():
+    with expected_protocol(
+            LeCroyT3DSO1204,
+            [("CHDR OFF", None),
+             (b"C1:BWL?", b"OFF"),
+             (b"BWL C1,OFF", None),
+             (b"C1:BWL?", b"ON"),
+             (b"BWL C1,ON", None)
+             ]
+    ) as instr:
+        assert instr.ch1.bwlimit is False
```

Comment:
It would be more natural to set it first and read it afterwards, but for the communication it is not that important.

Reply:
I changed that behavior wherever I could find it in unit tests.